src/blocks/NewQuizDialog.tsx

```diff
@@ -0,0 +1,135 @@
+import { Button, Dialog, DialogActions, DialogContent, FormControl, Grid, Slider, Typography } from '@mui/material';
+import { useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import { toast } from 'react-toastify';
+import { startQuizSession } from '../services/apiService';
+import { createQuiz } from '../services/lambdaService';
+
+interface NewQuizDialogProps {
+  open: boolean;
+  setOpen(open: boolean): void;
+  recordingId: string;
+}
+
+interface QuizSettings {
+  num_of_questions: number;
+  question_duration: number;
+}
+
+const NewQuizDialog = (props: NewQuizDialogProps) => {
+  const [quizSettings, setQuizSettings] = useState<QuizSettings>({ num_of_questions: 5, question_duration: 30 });
+  const handleChange = (e: Event) =>
+    setQuizSettings((prev) => ({
+      ...prev,
+      [(e.target as HTMLInputElement).name]: (e.target as HTMLInputElement).value,
+    }));
```

Comment:
Why do you think this is better than having two separate states and two handleChange functions?

Reply:
I suppose it's just a habit of mine; I tend to group related things (form fields in this case) into a single state variable. I find this a lot cleaner when there are a many fields since everything can be handled by a single state variable and a single function, but it doesn't make much of a difference here. It has no effect on performance, since the component will re-render any time either field changes, so it's ultimately just a code style preference.